iocage_lib/ioc_json.py

```diff
@@ -424,85 +455,98 @@ def _upgrade_pool(self, pool):
         self.zfs_set_property(pool, "org.freebsd.ioc:active", "yes")
         self.zfs_set_property(pool, "comment", "-")
 
-    def json_get_value(self, prop, default=False):
+    def json_get_value(self, prop):
         """Returns a string with the specified prop's value."""
-        old = False
-        zpools = list(map(lambda x: x.name, list(self.zfs.pools)))
-
-        if default:
-            _, iocroot = _get_pool_and_iocroot()
-            with open(f"{iocroot}/defaults.json", "r") as default_json:
-                conf = json.load(default_json)
-
-            if prop == "all":
-                return conf
-
-            return conf[prop]
-
         if prop == "pool":
-            match = 0
-
-            for pool in zpools:
+            raise DeprecationWarning("Access ioc_root.pool directly")
+        elif prop == "iocroot":
+            raise DeprecationWarning("Access ioc_root.iocroot directly")
+            return self.iocroot.mountpoint
+        elif prop == "all":
+            raise DeprecationWarning("Access the IOCJson instance directly")
```

Comment:
Why the deprecation warnings? Hardly a warning when it's fatal :P Also this interface is not deprecated in iocage, and this will break the existing FreeNAS middleware plugin.